File: Pset 6/readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")

    # Calling functions to get the letter, word, and sentence counts
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sent(text)

    # Coleman-Liau index calculation
    L = (letters * 100) / words
    S = (sentences * 100) / words
    index = round((0.0588 * L) - (0.296 * S) - 15.8)

    # Printing the appropriate reading level for the text
    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


# Defining funcion that inputs text and outputs the number of letters
def count_letters(string):
    let_count = 0
    for i in string:
        if i.isalpha():
            let_count += 1
    return let_count


# Defining funcion that inputs text and outputs the number of words
def count_words(string):
    word_count = 1
    for i in string:
        if i == " ":
            word_count += 1
    return word_count


# Defining funcion that inputs text and outputs the number of sentences
def count_sent(string):
    sent_count = 0
    for i in string:
        if i == "." or i == "!" or i == "?":
            sent_count += 1
    return sent_count


# Calling the main function to run the program
main()

File: Pset 2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

// Function Prototypes
int count_letters();
int count_words();
int count_sent();

int main(void)
{
    string s = get_string("Text: "); // Get a string from the user and save it as the variable s
    int let_count = count_letters(s); // Count the number of letters in the string s and save the result as let_count
    int word_count = count_words(s); // Count the number of words in the string s and save the result as word_count
    int sent_count = count_sent(s); // Count the number of sentences in the string s and save the result as sent_count
    
    float L = ((float) let_count * 100) / word_count; // Calculate the variable L in the Coleman-Liau index
    float S = ((float) sent_count * 100) / word_count; // Calculate the variable S in the Coleman-Liau index
    int index = round((0.0588 * L) - (0.296 * S) - 15.8); // Use L and S to calculate the index
    
    if (index < 1) // If the index result is less than 1, prunt out before Grade 1
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n"); // If the index result is greater than or equal to 16, prunt out Grade 16+
    }
    else
    {
        printf("Grade %i\n", index); // If the index is from 1 to 15, print out "Grade" followed by the index value
    }
}

int count_letters(char s[]) // Declare a funcion that inputs strings and outputs an integer
{
    int letters = 0; // Initialize a variable, letters, that keeps count of the number of letters in the string
    for (int i = 0, n = strlen(s); i < n; i++) // Run loop for as long as i is less than the length of the string
    {
        if ((s[i] >= 'a' && s[i] <= 'z') || (s[i] >= 'A' && s[i] <= 'Z')) // If the character in s is a letter, add 1 to letters
        {
            letters++;
        }
    }
    return letters; // Output the final number of letters
}

int count_words(char s[]) // Declare a funcion that inputs strings and outputs an integer
{
    int words = 0; // Initialize a variable, words, that keeps count of the number of words in the string
    for (int i = 0, n = strlen(s); i < n; i++) // Run loop for as long as i is less than the length of the string
    {
        if (s[i] == ' ') // If the character in s is a space, add 1 to words
        {
            words++;
        }
    }
    words++; // Add an additional 1 to words to account for the first word in the string
    return words; // Output the final number of words
}

int count_sent(char s[]) // Declare a funcion that inputs strings and outputs an integer
{
    int sentences = 0; // Initialize a variable, sentences, that keeps count of the number of sentences in the string
    for (int i = 0, n = strlen(s); i < n; i++) // Run loop for as long as i is less than the length of the string
    {
        if (s[i] == '.' || s[i] == '!' || s[i] == '?') // If the character in s is a typical sentence-ending character, add 1 to sentences
        {
            sentences++;
        }
    }
    return sentences; // Output the final number of sentences
}